sa4u_z3/main.py

```diff
@@ -249,29 +255,33 @@ def main():
     )
     parsed_args = parser.parse_args()
 
+    signal.signal(signal.SIGHUP, HUP_signal_handler)
+    
     _use_power_of_ten = parsed_args.power_of_ten
     _enable_scalar_prefixes = not parsed_args.disable_scalar_prefixes
 
     initialize_z3()
     tu_solver = solver
+    
+    while(True):
```

Comment:
nit: add a space after the while statement and expression, and the expression doesn't need to be in parens.